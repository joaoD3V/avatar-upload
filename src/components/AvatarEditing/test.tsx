import 'jest-canvas-mock';
import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helpers';

import { AvatarEditing } from '.';

const props = {
  src: 'https://99prod.s3.amazonaws.com/uploads/image/file/2429377/3eef648f6ce677d5e317509045df2be0.jpg',
};

describe('<AvatarEditing />', () => {
  it('should render without border and with close button by default', () => {
    renderWithTheme(<AvatarEditing {...props} />);

    expect(screen.queryByLabelText(/close button/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/Wrapper/i)).toHaveStyle({
      border: '0',
    });
  });

  it('should render avatar, crop slider and save button corretly', () => {
    const { container } = renderWithTheme(<AvatarEditing {...props} />);

    expect(screen.getByLabelText(/Avatar canva/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/Crop slider/i)).toBeInTheDocument();
    expect(screen.getByRole('button', { name: 'Save' })).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
