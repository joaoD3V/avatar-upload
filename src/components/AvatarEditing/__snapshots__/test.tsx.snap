// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `<AvatarEditing /> should render avatar, crop slider and save button corretly 1`
] = `
.c3 {
  width: 100%;
  height: -webkit-max-content;
  height: -moz-max-content;
  height: max-content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start !important;
  -webkit-box-align: flex-start !important;
  -ms-flex-align: flex-start !important;
  align-items: flex-start !important;
  -webkit-box-pack: start !important;
  -webkit-justify-content: flex-start !important;
  -ms-flex-pack: start !important;
  justify-content: flex-start !important;
}

.c4 {
  font-size: 400;
  font-size: 1.6rem;
  color: #677489;
}

.c5 {
  width: 100%;
  padding: 0 4px;
  height: -webkit-max-content;
  height: -moz-max-content;
  height: max-content;
  margin-top: 14px;
  position: relative !important;
}

.c5 .rc-slider {
  position: relative;
  width: 100%;
  height: 14px;
  padding: 5px 0;
  border-radius: 6px;
  touch-action: none;
  box-sizing: border-box;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
}

.c5 .rc-slider * {
  box-sizing: border-box;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
}

.c5 .rc-slider-rail {
  position: absolute;
  width: 100%;
  height: 2px;
  background-color: #a5c3ff;
  border-radius: 6px;
  pointer-events: none !important;
}

.c5 .rc-slider-track {
  position: absolute;
  height: 2px;
  background-color: #3F80FF;
  border-radius: 6px;
}

.c5 .rc-slider-handle {
  position: absolute;
  width: 12px;
  height: 12px;
  margin-top: -5px;
  background-color: #3F80FF;
  border-radius: 50%;
  cursor: pointer;
  cursor: -webkit--webkit-grab;
  cursor: -moz--webkit-grab;
  cursor: -webkit-grab;
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
  touch-action: pan-x;
}

.c5 .rc-slider-handle-dragging.rc-slider-handle-dragging.rc-slider-handle-dragging {
  border-color: #57c5f7;
}

.c5 .rc-slider-handle:focus {
  outline: none;
  box-shadow: none;
}

.c5 .rc-slider-handle:focus-visible {
  border-color: #2db7f5;
}

.c5 .rc-slider-handle-click-focused:focus {
  border-color: #96dbfa;
  box-shadow: unset;
}

.c5 .rc-slider-handle:hover {
  border-color: #57c5f7;
}

.c5 .rc-slider-handle:active {
  border-color: #57c5f7;
  cursor: -webkit--webkit-grabbing;
  cursor: -moz--webkit-grabbing;
  cursor: -webkit-grabbing;
  cursor: -webkit-grabbing;
  cursor: -moz-grabbing;
  cursor: grabbing;
}

.c5 .rc-slider-mark {
  position: absolute;
  top: 18px;
  left: 0;
  width: 100%;
  font-size: 12px;
}

.c5 .rc-slider-mark-text {
  position: absolute;
  display: inline-block;
  color: #999;
  text-align: center;
  vertical-align: middle;
  cursor: pointer;
}

.c5 .rc-slider-mark-text-active {
  color: #666;
}

.c5 .rc-slider-step {
  position: absolute;
  width: 100%;
  height: 2px;
  background: transparent;
  pointer-events: none;
}

.c5 .rc-slider-dot {
  position: absolute;
  bottom: -2px;
  width: 8px;
  height: 8px;
  vertical-align: middle;
  background-color: #fff;
  border: 2px solid #e9e9e9;
  border-radius: 50%;
  cursor: pointer;
}

.c5 .rc-slider-dot-active {
  border-color: #96dbfa;
}

.c5 .rc-slider-dot-reverse {
  margin-right: -4px;
}

.c5 .rc-slider-disabled {
  background-color: #e9e9e9;
}

.c5 .rc-slider-disabled .rc-slider-track {
  background-color: #ccc;
}

.c5 .rc-slider-disabled .rc-slider-handle,
.c5 .rc-slider-disabled .rc-slider-dot {
  background-color: #fff;
  border-color: #ccc;
  box-shadow: none;
  cursor: not-allowed;
}

.c5 .rc-slider-disabled .rc-slider-mark-text,
.c5 .rc-slider-disabled .rc-slider-dot {
  cursor: not-allowed !important;
}

.c5 .rc-slider-vertical {
  width: 14px;
  height: 100%;
  padding: 0 5px;
}

.c5 .rc-slider-vertical .rc-slider-rail {
  width: 4px;
  height: 100%;
}

.c5 .rc-slider-vertical .rc-slider-track {
  bottom: 0;
  left: 5px;
  width: 4px;
}

.c5 .rc-slider-vertical .rc-slider-handle {
  margin-top: 0;
  margin-left: -5px;
  touch-action: pan-y;
}

.c5 .rc-slider-vertical .rc-slider-mark {
  top: 0;
  left: 18px;
  height: 100%;
}

.c5 .rc-slider-vertical .rc-slider-step {
  width: 4px;
  height: 100%;
}

.c5 .rc-slider-vertical .rc-slider-dot {
  margin-left: -2px;
}

.c5 .rc-slider-tooltip-zoom-down-enter,
.c5 .rc-slider-tooltip-zoom-down-appear {
  display: block !important;
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation-play-state: paused;
  animation-play-state: paused;
}

.c5 .rc-slider-tooltip-zoom-down-leave {
  display: block !important;
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation-play-state: paused;
  animation-play-state: paused;
}

.c5 .rc-slider-tooltip-zoom-down-enter.rc-slider-tooltip-zoom-down-enter-active,
.c5 .rc-slider-tooltip-zoom-down-appear.rc-slider-tooltip-zoom-down-appear-active {
  -webkit-animation-name: rcSliderTooltipZoomDownIn;
  animation-name: rcSliderTooltipZoomDownIn;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c5 .rc-slider-tooltip-zoom-down-leave.rc-slider-tooltip-zoom-down-leave-active {
  -webkit-animation-name: rcSliderTooltipZoomDownOut;
  animation-name: rcSliderTooltipZoomDownOut;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c5 .rc-slider-tooltip-zoom-down-enter,
.c5 .rc-slider-tooltip-zoom-down-appear {
  -webkit-transform: scale(0,0);
  -ms-transform: scale(0,0);
  transform: scale(0,0);
  -webkit-animation-timing-function: cubic-bezier(0.23,1,0.32,1);
  animation-timing-function: cubic-bezier(0.23,1,0.32,1);
}

.c5 .rc-slider-tooltip-zoom-down-leave {
  -webkit-animation-timing-function: cubic-bezier(0.755,0.05,0.855,0.06);
  animation-timing-function: cubic-bezier(0.755,0.05,0.855,0.06);
}

.c5 .rc-slider-tooltip {
  position: absolute;
  top: -9999px;
  left: -9999px;
  visibility: visible;
  box-sizing: border-box;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
}

.c5 .rc-slider-tooltip * {
  box-sizing: border-box;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
}

.c5 .rc-slider-tooltip-hidden {
  display: none;
}

.c5 .rc-slider-tooltip-placement-top {
  padding: 4px 0 8px 0;
}

.c5 .rc-slider-tooltip-inner {
  min-width: 24px;
  height: 24px;
  padding: 6px 2px;
  color: #fff;
  font-size: 12px;
  line-height: 1;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #6c6c6c;
  border-radius: 6px;
  box-shadow: 0 0 4px #d9d9d9;
}

.c5 .rc-slider-tooltip-arrow {
  position: absolute;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
}

.c5 .rc-slider-tooltip-placement-top .rc-slider-tooltip-arrow {
  bottom: 4px;
  left: 50%;
  margin-left: -4px;
  border-width: 4px 4px 0;
  border-top-color: #6c6c6c;
}

.c0 {
  width: 553px;
  height: 177px;
  background: #F2F5F8;
  border: 0;
  border-radius: 8px;
  padding: 32px;
  position: relative;
}

.c7 {
  position: absolute;
  top: 32px;
  right: 32px;
  border: 0;
  outline: 0;
  background: transparent;
  cursor: pointer;
}

.c1 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c1 canvas {
  margin-right: 3.2rem;
  pointer-events: none;
}

.c2 {
  margin-top: 0.8rem;
  height: 100%;
  width: 276px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c6 {
  width: 109px;
  height: 32px;
  border-radius: 16px;
  background: #3D485F;
  color: #FFFFFF;
  border: 0;
  outline: 0;
  font-size: 1.4rem;
  font-weight: 500;
  margin-top: 35px;
  cursor: pointer;
  -webkit-transition: background-color 0.1s ease-in-out;
  transition: background-color 0.1s ease-in-out;
}

.c6:hover {
  background: #293040;
}

@media (max-width:768px) {
  .c0 {
    width: 90vw;
    height: 277px;
    padding: 16px;
  }
}

@media (max-width:768px) {
  .c1 canvas {
    margin-right: 0;
  }
}

@media (max-width:768px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:768px) {
  .c2 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<div
  aria-label="Wrapper"
  class="c0"
>
  <div
    class="c1"
  >
    <canvas
      aria-label="Avatar canva"
      height="113"
      style="width: 113px; height: 113px; cursor: grab;"
      width="113"
    />
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <span
          class="c4"
        >
          Crop
        </span>
        <div
          class="c5"
        >
          <div
            class="rc-slider rc-slider-horizontal"
          >
            <div
              class="rc-slider-rail"
            />
            <div
              class="rc-slider-track"
              style="left: 0%; width: 0%;"
            />
            <div
              class="rc-slider-step"
            />
            <div
              aria-disabled="false"
              aria-label="Crop slider"
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow="0"
              class="rc-slider-handle"
              role="slider"
              style="left: 0%; transform: translateX(-50%);"
              tabindex="0"
            />
          </div>
        </div>
      </div>
      <button
        class="c6"
        title="Save edit"
      >
        Save
      </button>
    </div>
  </div>
  <button
    aria-label="Close button"
    class="c7"
    title="Close edit"
  >
    <img
      alt="Icon with the symbol X, indicating the action to close edit"
      src="/img/close-icon.png"
    />
  </button>
</div>
`;
